src/auto/unix_socket_address.rs

```diff
@@ -0,0 +1,176 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use SocketAddress;
+use SocketConnectable;
+use UnixSocketAddressType;
+use ffi;
+use glib;
+use glib::StaticType;
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct UnixSocketAddress(Object<ffi::GUnixSocketAddress, ffi::GUnixSocketAddressClass>): SocketAddress, SocketConnectable;
+
+    match fn {
+        get_type => || ffi::g_unix_socket_address_get_type(),
+    }
+}
+
+impl UnixSocketAddress {
+    pub fn new(path: &str) -> UnixSocketAddress {
+        unsafe {
+            SocketAddress::from_glib_full(ffi::g_unix_socket_address_new(path.to_glib_none().0)).downcast_unchecked()
+        }
+    }
+
+    //pub fn new_abstract(path: /*Unimplemented*/&CArray TypeId { ns_id: 0, id: 10 }) -> UnixSocketAddress {
+    //    unsafe { TODO: call ffi::g_unix_socket_address_new_abstract() }
+    //}
+
+    //pub fn new_with_type(path: /*Unimplemented*/&CArray TypeId { ns_id: 0, id: 10 }, type_: UnixSocketAddressType) -> UnixSocketAddress {
+    //    unsafe { TODO: call ffi::g_unix_socket_address_new_with_type() }
+    //}
+
+    pub fn abstract_names_supported() -> bool {
+        unsafe {
+            from_glib(ffi::g_unix_socket_address_abstract_names_supported())
+        }
+    }
+}
+
+pub trait UnixSocketAddressExt {
+    fn get_address_type(&self) -> UnixSocketAddressType;
+
+    fn get_is_abstract(&self) -> bool;
+
+    fn get_path(&self) -> Option<String>;
```

Comment:
Is this always a proper string, i.e. utf8? Or can it also be a filesystem path for example? I think it can

Reply:
According to the [documentation](https://developer.gnome.org/gio/stable/GUnixSocketAddress.html#g-unix-socket-address-get-path) it can be both.

Reply:
That's inconvenient. Should return some enum then I guess, based on the type of the address?

Reply:
So, should it be a manual implementation?

Reply:
I guess so, returning the same enum you pass to the constructor probably.